testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/LoggedInPageHeaderTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.ui.account2;
+
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.testsuite.admin.ApiUtil;
+import org.keycloak.testsuite.ui.account2.page.AbstractLoggedInPage;
+import org.keycloak.testsuite.ui.account2.page.LoggedInPageHeaderPage;
+
+import static org.junit.Assert.assertTrue;
+
+public class LoggedInPageHeaderTest extends BaseAccountPageTest {
```

Comment:
I don't think we necessarily need to make a separate test for this. It could "piggyback" on some other test class, e.g. [PersonalInfoTest](https://github.com/keycloak/keycloak/blob/f2715025b344d37411701efec2df07376836979e/testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/PersonalInfoTest.java). (I think you had it like that at some point.)

But if you decide it should have its own test class (which is also ok), please don't base it on BaseAccountPageTest as it contains some features specific for a page test (like a navigation test, e.g.). You can use [AbstractAccountTest](https://github.com/keycloak/keycloak/blob/f2715025b344d37411701efec2df07376836979e/testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/AbstractAccountTest.java) instead as a much more suitable option.

Reply:
> @edewit Thanks for adding the test. I haven't yet tested it but all the changes in `PersonalInfoPage` should be moved to `LoggedInPageHeader`.

So you changed your mind again. I'll change the code again

Reply:
@edewit How did I change my mind? :) The citation you're mentioning is specifically about the page object whereas this thread is about the test class. If something's not clear please let me know – we could e.g. schedule a call and talk about it.

Reply:
Okay now I'm totally confused, based on you comment I moved all the stuff to separate classes. Then you comment that I shouldn't do that so I revert. I might have misinterpreted that, because you only mend the stuff from `PersonalInfoPage`, but here in this PR you also say that I shouldn't have a `LoggedInPageHeader`

Reply:
But I think I know what you meant now. Changed it.

Reply:
Thank you. My whole point was to simply follow the pattern we have in the tests to keep the code consistent and clean.